### ISSUE TITLE
TypeError when parsing template literal in dynamic import

### ISSUE THREAD

**Issue by mikecbrant** [eslint/eslint#12750]
**Tell us about your environment**

* **ESLint Version: _6.8.0_**
* **Node Version: _13.3.0_**
* **npm Version: _6.13.4_**

**What parser (default, Babel-ESLint, etc.) are you using?**

babel-eslint

**Please show your full configuration:**

**_.eslintrc.yml:_**
```
env:
  es2020: true
  browser: true
extends:
  - eslint:recommended
  - plugin:import/errors
  - plugin:import/warnings
  - plugin:security/recommended
  - plugin:react/recommended
parser: babel-eslint
parserOptions:
  ecmaVersion: 2020
  sourceType: module
  ecmaFeatures:
    impliedStrict: true
    jsx: true
plugins:
  - import
  - react
  - security
rules:
  indent:
    - error
    - 2
    - SwitchCase: 1
  linebreak-style:
    - error
    - unix
  quotes:
    - error
    - single
    - avoidEscape: true
  semi:
    - error
    - always
  # this rule is currently causing some false positives https://github.com/eslint/eslint/issues/11899
  # so we want to downgrade to warn so developer can take a look and fix if needed
  # if issue is fixed, we can remove this config
  require-atomic-updates: warn
  security/detect-eval-with-expression: error
  # this rule can flag cases where bracket notation access of objects is totally fine (i.e. no user input)
  # so we are downgrading this to a warn
  security/detect-object-injection: warn
settings:
  react:
    version: latest
```

**What did you do? Please include the actual source code causing the issue, as well as the command that you used to run ESLint.**

It seems that use of a template literal within a dynamic import is causing a `TypeError` to be thrown at parse time.  

```js
   // ... pertinent section of code below
    
    const promises = [
      import('intl-pluralrules'),
      import('@formatjs/intl-relativetimeformat/polyfill')
    ];
    supportedLocales.forEach(locale => {
      // the following line is causing parsing failure
      promises.push(import(`@formatjs/intl-relativetimeformat/dist/locale-data/${locale}`));
    });

    await Promise.all(promises);

```

It is run via a gulp-like task scheduler using gulp-eslint, so usage is in module context.

**What did you expect to happen?**

The template should parse correctly.

**What actually happened? Please include the actual, raw output from ESLint.**

`TypeError` is thrown when parsing module containing code sample above. Details shown below.

```
C:\...\node_modules\eslint\lib\rules\indent.js:340
        return this.setDesiredOffsets(token.range, fromToken, offset);
                                            ^
TypeError: Cannot read property 'range' of null
Occurred while linting C:\...\src\<omitted>.mjs:1
    at OffsetStorage.setDesiredOffset (C:\...\node_modules\eslint\lib\rules\indent.js:340:45)
    at C:\...\node_modules\eslint\lib\rules\indent.js:1358:29
    at Array.forEach (<anonymous>)
    at Object.TemplateLiteral [as listener] (C:\...\node_modules\eslint\lib\rules\indent.js:1350:34)
    at C:\.\node_modules\eslint\lib\rules\indent.js:1596:55
    at Array.forEach (<anonymous>)
    at Program:exit (C:\...\eslint\lib\rules\indent.js:1596:26)
    at C:\...\node_modules\eslint\lib\linter\safe-emitter.js:45:58
    at Array.forEach (<anonymous>)
    at Object.emit (C:\...\node_modules\eslint\lib\linter\safe-emitter.js:45:38)
npm ERR! Test failed.  See above for more details.
```

**Are you willing to submit a pull request to fix this bug?**

Potentially, if this issue ever becomes urgent for some reason.  Workaround is simple enough, simply fall back to more traditional string concatenation.  :)

**Comment by kaicataldo** [comment]
I believe this is an issue in `babel-eslint` (which should eventually be fixed by https://github.com/babel/babel/pull/10828). The default parser works fine with the `indent` rule (see demo [here](https://eslint.org/demo#eyJ0ZXh0IjoiaW1wb3J0KGBAZm9ybWF0anMvaW50bC1yZWxhdGl2ZXRpbWVmb3JtYXQvZGlzdC9sb2NhbGUtZGF0YS8ke2xvY2FsZX1gKSIsIm9wdGlvbnMiOnsicGFyc2VyT3B0aW9ucyI6eyJlY21hVmVyc2lvbiI6MTEsInNvdXJjZVR5cGUiOiJtb2R1bGUiLCJlY21hRmVhdHVyZXMiOnt9fSwicnVsZXMiOnsiaW5kZW50IjoyfSwiZW52Ijp7fX19)).
